src/helpers/figures/charts/runtime/chart_data_extractor.ts

```diff
@@ -275,6 +276,34 @@ export function getGeoChartData(
   };
 }
 
+export function getSunburstChartData(
+  definition: SunburstChartDefinition,
+  dataSets: DataSet[],
+  labelRange: Range | undefined,
+  getters: Getters
+): ChartRuntimeGenerationArgs {
+  // In Sunburst, labels are the leaf values (numbers), and the hierarchy is defined in the dataSets (strings)
+  let labels = getChartLabelValues(getters, dataSets, labelRange).values;
+  let dataSetsValues = getHierarchicalDatasetValues(getters, dataSets);
+  const removeFirstLabel = shouldRemoveFirstLabel(
+    labelRange,
+    dataSets[0],
+    definition.dataSetsHaveTitle || false
+  );
+  if (removeFirstLabel) {
+    labels.shift();
+  }
+  ({ labels, dataSetsValues } = filterInvalidHierarchicalPoints(labels, dataSetsValues));
+  ({ labels, dataSetsValues } = filterValuesWithDifferentSigns(labels, dataSetsValues));
```

Comment:
do you think we should switch those 2 lines : the complexity of `filterInvalidHierarchicalPoints` is bigger than the one of `filterValuesWithDifferentSigns`, so it might be faster to restrict first based on sign

Reply:
sure, it should be slightly better